module-system/sov-modules-macros/src/manifest.rs

```diff
@@ -236,6 +241,104 @@ impl<'a> Manifest<'a> {
         })
     }
 
+    pub fn parse_constant(
+        &self,
+        ty: &Type,
+        field: &Ident,
+        vis: syn::Visibility,
+        attrs: &[syn::Attribute],
+    ) -> Result<TokenStream, syn::Error> {
+        let map = self
```

Comment:
nit: this code is repeating with the `parse_gas_config`. Might be worthy to create a specialized, private scope function

```rust
fn get_object(&self, field: &Ident, key: &str) -> Result<TokenStream, syn::Error> {
  self
    .value
    .as_object()
    .ok_or_else(|| Self::err(&self.path, field, "manifest is not an object"))?
    .get(key)
    .ok_or_else(|| {
      Self::err(
        &self.path,
        field,
        format!("manifest does not contain a `{key}` attribute"),
      )
    })?
    .as_object()
    .ok_or_else(|| {
      Self::err(
        &self.path,
        field,
        format!("`{key}` attribute of `{field}` is not an object"),
      )
    })
}

Reply:
Good idea. Done!